sfdoc/publish/models.py

```diff
@@ -13,6 +13,8 @@
 
 class Article(models.Model):
     """Tracks created/updated articles per bundle."""
+    class Meta:
+        unique_together = [["bundle", "url_name"]]
```

Comment:
wouldn't it be valid for url_name to be duplicated if there was previously another article at the same URL that was deleted?

Reply:
You're right! It's really the filename which is the "unique key" per bundle.

Reply:
Okay I had to re-read the code to convince myself but I think my initial understanding was correct.

An article is "deleted" if it exists on PowerOfUs at url_name and no matching url_name exists in the HTML bundle on disk.

In other words: the url_name is actually the unique identifier for articles. For a moment I thought that it was filenames but that was images I was thinking about.

Reply:
Okay, I get it now.

Reply:
@prescod you are correct there should be no case where a 'deleted' (archived) article has a url_name matching a url_name in the html from the bundle.